src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -32,12 +32,22 @@ public ExpirationStorage(DocumentDatabase database, Transaction tx)
             tx.CreateTree(DocumentsByExpiration);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Put(DocumentsOperationContext context, Slice lowerId, BlittableJsonReaderObject document)
         {
             if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) == false ||
                 metadata.TryGet(Constants.Documents.Metadata.Expires, out string expirationDate) == false)
                 return;
 
+            PutInternal(context, lowerId, document);
+        }
+
+        private void PutInternal(DocumentsOperationContext context, Slice lowerId, BlittableJsonReaderObject document)
+        {
+            if (document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata) == false ||
```

Comment:
Why do double the work? Make this method accept the expiration date as parameter.

Reply:
Fixed